lib/tests/test_mut_repo.rs

```diff
@@ -715,3 +716,52 @@ fn test_reparent_descendants() {
         }
     }
 }
+
+#[test]
+fn test_bookmark_hidden_commit() {
+    // Test that MutableRepo::set_local_bookmark_target() on a hidden commit updates
+    // the view head ids as setting it makes the commit visible.
+    let test_repo = TestRepo::init();
+    let repo = &test_repo.repo;
+    let root_commit = repo.store().root_commit();
+
+    let mut tx = repo.start_transaction();
+    let wc_commit = write_random_commit(tx.repo_mut());
+
+    // Intentionally not doing tx.commit, so the commit id is not tracked
+    // in the view head ids.
+
+    let mut tx = repo.start_transaction();
+    tx.repo_mut()
+        .set_local_bookmark_target("b", RefTarget::normal(wc_commit.id().clone()));
+    let repo = tx.commit("test").unwrap();
+    assert_eq!(
+        *repo.view().heads(),
+        hashset! {wc_commit.id().clone(), root_commit.id().clone()}
+    );
+}
+
+#[test]
+fn test_bookmark_hidden_predecessor() {
```

Comment:
This looks like effectively the same test case as the one above (you just create an extra `wc_id` variable that's unused AFAICT). I don't think we need a separate test for unhiding a predecessor, so I would just delete this test.

Reply:
I copied it because I wanted to write a second test, but it is gone now.

Reply:
should be fixed now